src/idl_gen_general.cpp

```diff
@@ -731,6 +731,7 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
        it != struct_def.fields.vec.end();
        ++it) {
     auto &field = **it;
+    assert(field.value.type.base_type != BASE_TYPE_ARRAY);
```

Comment:
Function for struct generation is out of generator class in go and python generators, so this function has not access to the `error_` field. I see two solutions here: can pass `error_` as additional parameter to `GenStruct` function or wrap `GenStruct` and `GenEnum` into generator class.